src/targets.rs

```diff
@@ -81,6 +81,70 @@ field_impls!(Os);
 field_impls!(Family);
 field_impls!(Env);
 
+/// A set of families for a target.
+///
+/// Each target can be part of one or more families. This struct represents them.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Families(Cow<'static, [Family]>);
+
+impl Families {
+    /// Constructs a new instance.
+    ///
+    /// This method accepts both owned `String`s and `&'static str`s.
+    ///
+    /// If you have a `&'static [&'static str]`, prefer [`Self::new_const`].
+    #[inline]
+    pub fn new(val: impl IntoIterator<Item = Family>) -> Self {
+        let mut families: Vec<_> = val.into_iter().collect();
+        families.sort_unstable();
+        Self(Cow::Owned(families))
+    }
+
+    /// Constructs a new instance of this field from a static slice of `&'static str`.
+    ///
+    /// `val` must be in sorted order: this constructor cannot check for that due to
+    /// limitations in current versions of Rust.
+    #[inline]
+    pub const fn new_const(val: &'static [Family]) -> Self {
+        // TODO: Check that val is sorted.
+        Self(Cow::Borrowed(val))
+    }
+
+    /// Returns true if this list of families contains a given family.
+    pub fn contains(&self, val: &Family) -> bool {
+        // This relies on the sorted-ness of its contents.
+        self.0.binary_search(val).is_ok()
+    }
+}
```

Comment:
Given that the maximum for a reasonable target is 2, at least at this time, it feels like sorting/binary searching is unnecessary and should just be linear.

Reply:
Thanks! Note that while the binary searching is probably unnecessary, the sorting is not: incorrectly sorted lists will cause the `Eq`, `Hash` and `Ord` implementations to behave incorrectly.